projects/aca-content/src/lib/components/knowledge-retrieval/search-ai/search-ai-input/search-ai-input.component.ts

```diff
@@ -128,11 +128,16 @@ export class SearchAiInputComponent implements OnInit, OnDestroy {
     private userPreferencesService: UserPreferencesService,
     private translateService: TranslateService,
     private modalAiService: ModalAiService,
-    private route: ActivatedRoute
+    private route: ActivatedRoute,
+    private router: Router
   ) {}
 
   ngOnInit(): void {
-    if (this.searchTerm) {
+    const isKnowledgeRetrievalPage = this.router.url.startsWith('/knowledge-retrieval');
```

Comment:
I have one idea to avoid using url here because it would be safer to not use urls, just in case they are changed because when somebody changes url for KR then it can break this place. 

Above we have following input: 
useStoredNodes

This input is set only for search results to true, for other places it's not true. Can we rename this input to something like: "usedInAiResultsPage"? So then this flag can be used here instead of url as it will be true only for KR results page. This seems safer because it's easier to break something when string changes so if anyone changes url then it will be harder to detect